massa-execution-exports/src/execution_info.rs

```diff
@@ -0,0 +1,343 @@
+#![allow(dead_code)]
+
+use std::collections::HashMap;
+
+use massa_async_pool::AsyncMessageId;
+use massa_hash::Hash;
+use massa_models::block_id::BlockId;
+use massa_models::config::{GENESIS_TIMESTAMP, T0, THREAD_COUNT};
+use massa_models::denunciation::DenunciationIndex;
+use massa_time::MassaTime;
+use schnellru::{ByLength, LruMap};
+
+use massa_deferred_calls::DeferredCall;
+use massa_models::address::Address;
+use massa_models::amount::Amount;
+use massa_models::slot::Slot;
+use serde::Serialize;
+
+use crate::types_trace_info::ExecutionResult;
+// #[cfg(feature = "execution-trace")]
+// use crate::Transfer;
+
+/// Struct for Execution info per slot
+pub struct ExecutionInfo {
+    /// Map of execution info
+    pub info_per_slot: LruMap<Slot, ExecutionInfoForSlot>,
+}
+
+impl ExecutionInfo {
+    /// Create a new ExecutionInfo
+    pub fn new(max_slot_size_cache: u32) -> Self {
+        Self {
+            info_per_slot: LruMap::new(ByLength::new(max_slot_size_cache)),
```

Comment:
what is the purpose of this cache ? Can't we just store the ExecutionInfoForSlot at the speculative slot level, and simply broadcast it on the tokio broadcast when the slot finalizes ?

Reply:
The cache was originaly there to be able to query it usign grpc call like: get_exec_info_for_slot (but it has not been impl yet)

Reply:
Done

Reply:
not resolved